### ISSUE TITLE
Linkerd identity Issue upgrading from 2.11 to 2.12

### ISSUE THREAD

**Issue by irizzant** [linkerd/linkerd2#9398]
### What is the issue?

I have linkerd with version 2.11.2 along with viz and Jaeger extensions installed via Helm.

When I try upgrading to 2.12.0 using the new charts linkerd identity pods don't start correctly, and they report the following error:
![image](https://user-images.githubusercontent.com/15982257/190101867-ff05d640-72cb-441d-8573-afeac3864b95.png)

As per my understanding the error is exactly the one reported in https://github.com/linkerd/linkerd2/issues/6740

### How can it be reproduced?

Install linkerd version 2.11.2 along with viz and Jaeger extensions via Helm and try the upgrade

### Logs, error output, etc

![image](https://user-images.githubusercontent.com/15982257/190101867-ff05d640-72cb-441d-8573-afeac3864b95.png)

### output of `linkerd check -o short`

```text
Linkerd core checks
===================

linkerd-identity
----------------
‼ issuer cert is valid for at least 60 days
    issuer certificate will expire on 2022-09-16T04:27:41Z
    see https://linkerd.io/2.11/checks/#l5d-identity-issuer-cert-not-expiring-soon for hints

linkerd-webhooks-and-apisvc-tls
-------------------------------
‼ proxy-injector cert is valid for at least 60 days
    certificate will expire on 2022-09-14T10:38:25Z
    see https://linkerd.io/2.11/checks/#l5d-proxy-injector-webhook-cert-not-expiring-soon for hints
‼ sp-validator cert is valid for at least 60 days
    certificate will expire on 2022-09-14T10:38:28Z
    see https://linkerd.io/2.11/checks/#l5d-sp-validator-webhook-cert-not-expiring-soon for hints
‼ policy-validator cert is valid for at least 60 days
    certificate will expire on 2022-09-14T22:52:33Z
    see https://linkerd.io/2.11/checks/#l5d-policy-validator-webhook-cert-not-expiring-soon for hints

linkerd-version
---------------
‼ cli is up-to-date
    is running version 2.11.2 but the latest stable version is 2.12.0
    see https://linkerd.io/2.11/checks/#l5d-version-cli for hints

control-plane-version
---------------------
‼ control plane is up-to-date
    is running version 2.11.2 but the latest stable version is 2.12.0
    see https://linkerd.io/2.11/checks/#l5d-version-control for hints

linkerd-control-plane-proxy
---------------------------
‼ control plane proxies are up-to-date
    some proxies are not running the current version:
        * linkerd-destination-f8fc8d85-2dbwp (stable-2.11.2)
        * linkerd-destination-f8fc8d85-6cscm (stable-2.11.2)
        * linkerd-destination-f8fc8d85-jgxp9 (stable-2.11.2)
        * linkerd-identity-5969fb88d4-47v4p (stable-2.11.2)
        * linkerd-identity-5969fb88d4-bv8n7 (stable-2.11.2)
        * linkerd-identity-5969fb88d4-fvw6s (stable-2.11.2)
        * linkerd-proxy-injector-5994874fb7-4b54q (stable-2.11.2)
        * linkerd-proxy-injector-5994874fb7-6r7g4 (stable-2.11.2)
        * linkerd-proxy-injector-5994874fb7-b66fm (stable-2.11.2)
    see https://linkerd.io/2.11/checks/#l5d-cp-proxy-version for hints

Linkerd extensions checks
=========================

linkerd-jaeger
--------------
‼ jaeger extension proxies are up-to-date
    some proxies are not running the current version:
        * jaeger-injector-6d5df5bc95-xlfgg (stable-2.11.2)
    see https://linkerd.io/2.11/checks/#l5d-jaeger-proxy-cp-version for hints

Linkerd extensions checks
=========================

linkerd-viz
-----------
‼ tap API server cert is valid for at least 60 days
    certificate will expire on 2022-09-15T07:35:41Z
    see https://linkerd.io/2.11/checks/#l5d-tap-cert-not-expiring-soon for hints
‼ viz extension proxies are up-to-date
    some proxies are not running the current version:
        * grafana-84cd44d9b9-v2zf8 (stable-2.11.2)
        * metrics-api-6c5844ddcc-rd8m4 (stable-2.11.2)
        * prometheus-894fb5bd7-c4vk8 (stable-2.11.2)
        * tap-755d5dbb64-2mjl5 (stable-2.11.2)
        * tap-755d5dbb64-p8249 (stable-2.11.2)
        * tap-755d5dbb64-whg8l (stable-2.11.2)
        * tap-injector-569d59cfbc-w52r2 (stable-2.11.2)
        * web-6884745749-7zc6m (stable-2.11.2)
    see https://linkerd.io/2.11/checks/#l5d-viz-proxy-cp-version for hints

Status check results are √

```

### Environment

- Kubernetes version 
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.0", GitCommit:"ab69524f795c42094a6630298ff53f3c3ebab7f4", GitTreeState:"clean", BuildDate:"2021-12-07T18:16:20Z", GoVersion:"go1.17.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.8", GitCommit:"7061dbbf75f9f82e8ab21f9be7e8ffcaae8e0d44", GitTreeState:"clean", BuildDate:"2022-03-16T14:04:34Z", GoVersion:"go1.16.15", Compiler:"gc", Platform:"linux/amd64"}

**Comment by irizzant** [comment]
Adding some context here.

I've deployed linkerd using the [gitops approach](https://linkerd.io/2.12/tasks/gitops/) with ArgoCD and followed the [guide for upgrading](https://linkerd.io/2.12/tasks/upgrade/#upgrade-notice-stable-2-12-0) to linkerd 2.12 from 2.11with helm.

At the moment I found no way to successfully upgrade linkerd to 2.12.0 without deleting and then recreating the control plane deployments, which causes downtime.
